runners/cli/src/main/kotlin/cli/main.kt

```diff
@@ -406,14 +404,35 @@ fun parseLinks(links: List<String>): List<ExternalDocumentationLink> {
             }
 }
 
+fun initializeConfiguration(globalArguments: GlobalArguments): DokkaConfiguration = if (globalArguments.json != null) {
+        val jsonContent = Paths.get(checkNotNull(globalArguments.json)).toFile().readText()
+        val globals: GlobalDokkaConfiguration = parseJson(jsonContent)
+        val dokkaConfigurationImpl = DokkaConfigurationImpl(jsonContent)
+        dokkaConfigurationImpl.run {
+            sourceSets.forEach {
+                it.perPackageOptions.cast<MutableList<DokkaConfiguration.PackageOptions>>().addAll(globals.perPackageOptions ?: emptyList())
+            }
+
+            sourceSets.forEach {
+                it.externalDocumentationLinks.cast<MutableSet<ExternalDocumentationLink>>().addAll(globals.externalDocumentationLinks ?: emptyList())
+            }
+
+            sourceSets.forEach {
+                it.sourceLinks.cast<MutableSet<SourceLinkDefinitionImpl>>().addAll(globals.sourceLinks ?: emptyList())
+            }
+
+            sourceSets.forEach {
+                it.externalDocumentationLinks.cast<MutableSet<ExternalDocumentationLink>>().addAll(defaultLinks(it))
+            }
+        }
```

Comment:
it seems that very similar logic is present in [GlobalArguments's init](https://github.com/Kotlin/dokka/blob/0ae0222561d86a44325e17c0296cfc87f93f380e/runners/cli/src/main/kotlin/cli/main.kt#L133), what do you think about moving the contents of this method there for consistency?

Reply:
`init` method is used for `cli` parameters while these settings are obtained from json file, which is processed as a completely different path of execution (it always has been since this check: 
```
    val configuration = if (globalArguments.json != null)
        DokkaConfigurationImpl(
            Paths.get(checkNotNull(globalArguments.json)).toFile().readText()
        )
    else
        globalArguments
```
)
I assume you cannot both have some settings setup as `cli` params and some passed via JSON, so I think it's not worth it trying to unify these two completely paths of execution, but I am open to your opinions